desktop/apps/artwork/routes.coffee

```diff
@@ -69,6 +69,7 @@ bootstrap = ->
   inPurchaseTestGroup = res.locals.sd.PURCHASE_FLOW is 'purchase'
   metaphysics send
     .then (data) ->
+      return next() unless data.artwork?
```

Comment:
So if you go to `/artwork/invalid`, the GraphQL query is successful, but there's really no data since the artwork request itself fails. By not rejecting the promise anymore, we get an ugly crash about referencing some non existent data here, so this is the 'correct' fix. ie- this artwork app makes a determination about how much data is enough and can `next()` otherwise.